exams/pearson/audit_test.py

```diff
@@ -0,0 +1,281 @@
+    settings[request.param] = None
+
+    with override_settings(**settings):
+        yield settings
+
+
+@pytest.fixture()
+def s3_store(valid_settings):
+    """S3 storage for tests"""
+    return audit.S3AuditStorage()
+
+
+@pytest.fixture
+def auditor(valid_settings):
+    """
+    ExamDataAuditor that cleans up the GPG keys after itself
+    """
+    gpg = gnupg.GPG()
+    yield audit.ExamDataAuditor(gpg=gpg, store=Mock(spec=audit.S3AuditStorage))
+    gpg.delete_keys(FINGERPRINT_STRIPPED)
+
+
+def test_s3_store_get_bucket(s3_store, mocker):
+    """Test get_bucket"""
+    with override_settings(**{
+        'EXAMS_AUDIT_S3_BUCKET': None,
+    }):
+        with pytest.raises(ImproperlyConfigured):
+            s3_store.get_bucket()
+
+
+def test_s3_store_get_connection_valid(s3_store, mocker):
+    """Test get_connection given valid settings"""
+    mock = mocker.patch('boto.s3.connection.S3Connection')
+    s3_store.get_bucket()
+    mock.assert_called_once_with('not.an.id', 'not.an.access.key')
+
+
+@pytest.mark.parametrize('key', [
+    'EXAMS_AUDIT_AWS_ACCESS_KEY_ID',
+    'EXAMS_AUDIT_AWS_SECRET_ACCESS_KEY',
+])
+def test_s3_store_get_connection(s3_store, key):
+    """Test get_connection against missing settings"""
+    with override_settings(**{
+        key: None,
+    }):
+        with pytest.raises(ImproperlyConfigured):
+            s3_store.get_connection()
+
+
+def test_s3_store_get_s3_key(s3_store, mocker):
+    mocker.patch.object(s3_store, 'get_bucket', return_value='.the.bucket.')
+    key = s3_store.get_s3_key('filename', 'filetype')
+    assert key.bucket == s3_store.get_bucket.return_value
+    assert key.key == 'exam_audits/filetype/filename'
+    s3_store.get_bucket.assert_called_once_with()
+
+
+def test_exam_data_auditor_fingerprint(auditor):
+    """Test that fingerprint strips whitespace from the key"""
+    assert auditor.fingerprint == FINGERPRINT_STRIPPED
+
+
+def test_exam_data_auditor_public_key(auditor):
+    """Test that this decodes a base64 encoded pubkey"""
+    assert auditor.public_key == PUBLIC_KEY
+
+
+def test_exam_data_auditor_configure(auditor):
+    """Test that configure() succeeds with correct settings"""
+    auditor.configure()
+    assert auditor.gpg.list_keys().fingerprints == [FINGERPRINT_STRIPPED]
+    assert auditor.configured is True
+
+
+def test_exam_data_auditor_configure_already_configured(auditor, mocker):
+    """Test that configure() succeeds if already configured"""
+    mocker.patch.object(auditor, 'get_fingerprints', return_value=[FINGERPRINT_STRIPPED])
+    mocker.spy(auditor.gpg, 'import_keys')
+    auditor.configure()
+    assert auditor.gpg.import_keys.call_count == 0
+    assert auditor.configured is True
+
+
+def test_exam_data_auditor_enabled(auditor, missing_settings, mocker):
+    """Test that audit() respected the enabled flag"""
+    mocker.patch.object(auditor, 'configure')
+    mocker.patch.object(auditor, 'upload_encrypted_file')
```

Comment:
can you use `patch.multiple` instead of 2 `patch.object` calls here? http://www.voidspace.org.uk/python/mock/patch.html#patch-multiple